### ISSUE TITLE
Direction key's overlapping hierarchy

### ISSUE THREAD

**Issue by jsnetors** [chrismaltby/gb-studio#433]
Don't know if a bug, but if you're holding the vertical direction key and presses an horizontal one at the same time, the latter replaces the previous, but it doesn't happen when it's an horizontal key and you press up or down.

**Comment by RichardULZ** [comment]
Just an order of button check, not sure if other nes or gb games with 4 directions handle this any better. 
I think we just check horizontal keys or whatever key first, and if it is currently pressed, branch off from there. 

We do have a last frame joypad state, but i think that's for other things, attach script to button is checked before so overrides and fires once per hold (unless last joypad is reset, eg by a trigger or timer script?), but again in a similar order if many buttons are held down.

**Comment by rorosaurus** [comment]
Yeah, I noticed this too. It's easiest to see when playing an HTML5 game, or emulating ROM, where you aren't using an actual D-pad that prevents opposite button presses. But even with an actual D-pad, for example, if you hold left, then quick press down - the down input will be suppressed.

The button priority appears to be: left, right, up, down.

I think the ideal behavior would be to use the latest-held-down button (if only one directional input is used). Or use the latest-held-down-vertical button AND latest-held-down-horizontal button (if diagonal input is valid for your game).
